working/0508 - strict-inference/strict-inference.md

```diff
@@ -0,0 +1,362 @@
+# Strict inference static analysis option
```

Comment:
I don't think working/ is the right sub-directory for this long term.  The usual model is that working is for us to noodle and iterate in, then things moved to accepted/ and then from there to 2.x/ .  If that's what you have in mind for this, that's fine, but I think it probably makes most sense to add this to resources/type-system .  WDYT?